src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -5042,6 +5042,279 @@ class StubGenerator: public StubCodeGenerator {
+   */
+  address generate_updateBytesAdler32() {
+    __ align(CodeEntryAlignment);
+    StubCodeMark mark(this, "StubRoutines", "updateBytesAdler32");
+    address start = __ pc();
+
+    Label L_simple_by1_loop, L_nmax, L_nmax_loop, L_nmax_loop_entry,
+      L_by16, L_by16_loop, L_by16_loop_unroll, L_by1_loop, L_do_mod, L_combine, L_by1;
+
+    // Aliases
+    Register adler  = c_rarg0;
+    Register s1     = c_rarg0;
+    Register s2     = c_rarg3;
+    Register buff   = c_rarg1;
+    Register len    = c_rarg2;
+    Register nmax  = x29; // t4
+    Register base  = x30; // t5
+    Register count = x31; // t6
+    Register temp0 = c_rarg4;
+    Register temp1 = c_rarg5;
+    Register temp2 = c_rarg6;
+    Register temp3 = x28; // t3
+
+    VectorRegister vzero = v31;
+    VectorRegister vbytes = v8; // group: v8, v9, v10, v11
+    VectorRegister vs1acc = v12; // group: v12, v13, v14, v15
+    VectorRegister vs2acc = v16; // group: v16, v17, v18, v19, v20, v21, v22, v23
+    VectorRegister vtable_64 = v24; // group: v24, v25, v26, v27
+    VectorRegister vtable_32 = (MaxVectorSize == 16) ? v26 : v4; // group: v4, v5
+    VectorRegister vtable_16 = (MaxVectorSize == 16) ? v27 : v30;
+    VectorRegister vtemp1 = v28;
+    VectorRegister vtemp2 = v29;
+
+    // Max number of bytes we can process before having to take the mod
+    // 0x15B0 is 5552 in decimal, the largest n such that 255n(n+1)/2 + (n+1)(BASE-1) <= 2^32-1
+    const uint64_t BASE = 0xfff1;
+    const uint64_t NMAX = 0x15B0;
+
+    // Loops steps
+    int step_64 = 64;
+    int step_32 = 32;
+    int step_16 = 16;
+    int step_1  = 1;
+
+    __ enter(); // Required for proper stackwalking of RuntimeStub frame
+    __ mv(temp1, 64);
+    __ vsetvli(temp0, temp1, Assembler::e8, Assembler::m4);
+
+    // Generating accumulation coefficients for further calculations
+    // vtable_64:
+    __ vid_v(vtemp1);
+    __ vrsub_vx(vtable_64, vtemp1, temp1);
+    // vtable_64 group now contains { 0x40, 0x3f, 0x3e, ..., 0x3, 0x2, 0x1 }
+
+    if (MaxVectorSize > 16) {
+      // Need to generate vtable_32 explicitly
+      __ mv(temp1, 32);
+      __ vsetvli(temp0, temp1, Assembler::e8, Assembler::m2);
+      __ vid_v(vtemp1);
+      __ vrsub_vx(vtable_32, vtemp1, temp1);
+      // vtable_32 group now contains { 0x20, 0x1f, 0x1e, ..., 0x3, 0x2, 0x1 }
+    }
+
+    __ vsetivli(temp0, 16, Assembler::e8, Assembler::m1);
+    if (MaxVectorSize > 16) {
+      // Need to generate vtable_16 explicitly
+      __ mv(temp1, 16);
+      __ vid_v(vtemp1);
+      __ vrsub_vx(vtable_16, vtemp1, temp1);
+      // vtable_16 group now contains { 0x10, 0xf, 0xe, ..., 0x3, 0x2, 0x1 }
+    }
+
+    __ vmv_v_i(vzero, 0);
+
+    __ mv(base, BASE);
+    __ mv(nmax, NMAX);
+
+    // s1 is initialized to the lower 16 bits of adler
+    // s2 is initialized to the upper 16 bits of adler
+    __ zero_extend(s1, s1, 16); // s1 = (adler & 0xffff)
+    __ srliw(s2, adler, 16); // s2 = ((adler >> 16) & 0xffff)
```

Comment:
This doesn't seem corrent to me. Note that `s1` is an alias of `adler` at the entry. 
```
    Register adler  = c_rarg0;
    Register s1     = c_rarg0;
```

Reply:
Thanks! Just in case I missed something else: i re-ran the `test/hotspot/jtreg/compiler/intrinsics/zip/TestAdler32.java` and it's OK.